### DISCUSSION TITLE
Crow (v4) and Dove (v5) backward compatibility

### DISCUSSION THREAD

**Discussion by jamesholcomb** [feathersjs/feathers#2443] [Q&A] [answered]
Are v4 clients (socket and REST) compatible with v5 servers (and vice versa)?

**Comment by jamesholcomb** [comment]
Wanted to bring this up since there were a couple of gotchas/surprises with the last upgrade with respect to compat.

**Comment by daffl** [comment, marked as answer]
The message and request format is the same, however Socket.io introduced breaking protocol changes that later got smoothed out with an explicit backwards compatibility opt in:

```js
app.configure(socketio({
  allowEIO3: true // false by default
});
```

More information can be found at https://socket.io/docs/v3/migrating-from-2-x-to-3-0/index.html#How-to-upgrade-an-existing-production-deployment (also linked from the migration guide).
